velox/functions/sparksql/tests/ArrayInsertTest.cpp

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+
+class ArrayInsertTest : public SparkFunctionBaseTest {
+ protected:
+  void SetUp() override {
+    SparkFunctionBaseTest::SetUp();
+    options_.parseIntegerAsBigint = false;
+  }
+
+  void testExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    const auto result = evaluate(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  void testExpression(
+      const std::string& expression,
+      const RowVectorPtr& input,
+      const VectorPtr& expected) {
+    const auto result = evaluate(expression, input);
+    assertEqualVectors(expected, result);
+  }
+
+  void evaluateExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    evaluate(expression, makeRowVector(input));
+  }
+
+  template <typename T>
+  void testInt() {
+    const auto input = makeRowVector({
+        makeArrayVectorFromJson<T>({"[1]", "[2, 2]"}),
+        makeNullableFlatVector<T>({0, std::nullopt}),
+    });
+
+    auto expected = makeArrayVectorFromJson<T>({"[0, 1]", "[null, 2, 2]"});
+    testExpression("array_insert(c0, 1, c1, false)", input, expected);
+
+    expected = makeArrayVectorFromJson<T>({"[0, 1]", "[2, null, 2]"});
+    testExpression("array_insert(c0, -2, c1, false)", input, expected);
+  }
+
+  template <typename T>
+  void testFloat() {
+    const auto input = makeRowVector({
+        makeNullableArrayVector<T>(
+            {{1.0001, -2.0},
+             {2.0001, std::numeric_limits<T>::infinity()},
+             {std::numeric_limits<T>::quiet_NaN(), 9.0009},
+             {3.0001}}),
+        makeNullableFlatVector<T>(
+            {0,
+             std::nullopt,
+             std::numeric_limits<T>::infinity(),
+             std::numeric_limits<T>::quiet_NaN()}),
+    });
+
```

Comment:
Perhaps define the special values as variables and reuse them like below.

https://github.com/facebookincubator/velox/blob/58a80e5c297b5c273e56c9f9ba9b9956fbeeb075/velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp#L1452-L1454

Reply:
Thanks, done.